src/main/resources/static/templates/PersistentXSSInHTMLTagVulnerability/LEVEL_1/PersistentXSS.js

```diff
@@ -14,13 +17,33 @@ function addingEventListenerToLoadImageButton() {
 }
 
 function appendComment(post) {
-  postDiv.innerHTML = post.content;
+  // Create a new div element to hold the post content
+  let postDiv = document.createElement('div');
+  
+  // Use textContent to safely assign user-controlled data
+  postDiv.textContent = post.content;
+  
+  // Append the safe postDiv to the allPosts element
   document.getElementById("allPosts").appendChild(postDiv);
 }
 
 function appendResponseCallback(data) {
-  document.getElementById("allPosts").innerHTML = data;
```

Comment:
CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')
------
User controlled data in methods like `innerHTML`, `outerHTML` or `document.write` is an anti-pattern that can lead to XSS vulnerabilities